lib/splitclient-rb/engine/matchers/in_list_semver_matcher.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module SplitIoClient
+  class InListSemverMatcher < Matcher
+    MATCHER_TYPE = 'IN_LIST_SEMVER'
+
+    attr_reader :attribute
+
+    def initialize(attribute, list_value, logger, validator)
+      super(logger)
+      @validator = validator
+      @attribute = attribute
+      @semver_list = list_value.map { |item| SplitIoClient::Semver.build(item, logger) }
+      @logger = logger
+    end
+
+    def match?(args)
+      return false unless verify_semver_arg?(args, 'InListSemverMatcher')
+
+      value_to_match = SplitIoClient::Semver.build(args[:attributes][@attribute.to_sym], @logger)
+      unless !value_to_match.nil? && @semver_list.all? { |n| !n.nil? }
```

Comment:
and you can check only `value_to_matc` here, eg:`if value_to_match.nil?`

Reply:
will update